rosbag2_test_common/include/rosbag2_test_common/temporary_directory_fixture.hpp

```diff
@@ -38,64 +31,15 @@ class TemporaryDirectoryFixture : public Test
 public:
   TemporaryDirectoryFixture()
   {
-    char template_char[] = "tmp_test_dir.XXXXXX";
-#ifdef _WIN32
-    char temp_path[255];
-    GetTempPathA(255, temp_path);
-    _mktemp_s(template_char, strnlen(template_char, 20) + 1);
-    temporary_dir_path_ = std::string(temp_path) + std::string(template_char);
-    _mkdir(temporary_dir_path_.c_str());
-#else
-    char * dir_name = mkdtemp(template_char);
-    temporary_dir_path_ = dir_name;
-#endif
+    auto tmpdir = rcpputils::fs::temp_directory_path();
+    temporary_dir_path_ = rcpputils::fs::create_temp_directory(
+      rcpputils::fs::temp_directory_path(),
```

Comment:
as mentioned in the respective `rcpputils` PR, I somewhat feel like this API is redundant. I've actually reviewed the other one first, but looking at this now, what about changing the `create_temp_directory` function to default to `rcpputils::fs::temp_directory_path()`?

```
path create_temp_directory(const std::string & base_name, const path & parent = rcpputils::fs::create_temp_directory())
```

Reply:
That default seems like a good idea, I'll update it